src/codeu/chat/server/MessagePersist.java

```diff
@@ -0,0 +1,182 @@
+		}
+		catch (Exception e) {
+			System.err.println( e.getClass().getName() + ": " + e.getMessage() );
+      		System.exit(0);			
+		}
+	}
+
+	public void retrieveUser(Controller controller) {
+		try {
+			ResultSet rs = stmt.executeQuery( "SELECT * FROM USERS;");
+			while (rs.next() ) {
+				Uuid id = Uuid.fromString(rs.getString("id"));
+				String name = rs.getString("name");
+				Time time = Time.fromMs(rs.getLong("creationtime"));
+
+				controller.newUser(id, name, time);
+			}
+			rs.close();
+			stmt.close();
+		}
+		catch ( Exception e ) {
+      		System.err.println( e.getClass().getName() + ": " + e.getMessage() );
+      		System.exit(0);
+   		}
+	}
+
+	public void createConversationTable() {
+		try {
+			String sql = "CREATE TABLE IF NOT EXISTS CONVERSATION " + 
+						 "(ID            TEXT      NOT NULL," + 
+						 " TITLE         TEXT      NOT NULL," + 
+						 " OWNER         TEXT      NOT NULL," + 
+						 " CREATIONTIME  INTEGER      NOT NULL)";
+			stmt.executeUpdate(sql);
+			stmt.close();
+		}
+		catch (Exception e) {
+			System.err.println( e.getClass().getName() + ": " + e.getMessage() );
+    		System.exit(0);
+    	}
+    }
+
+	public void insertConversationTable(String id, String title, String ownerID, Long time) {
+		try {
+			String sql = "INSERT INTO CONVERSATION (ID, TITLE, OWNER, CREATIONTIME)" + 
+						 "VALUES ( '" + id + "' , '" + title + "' , '" + ownerID + "' , '" + time + "');" ;
+			stmt.executeUpdate(sql);
+			stmt.close();
+		}
+		catch (Exception e) {
+			System.err.println( e.getClass().getName() + ": " + e.getMessage() );
+      		System.exit(0);			
+		}
+
+	}
+
+	public void retrieveConversation(Controller controller) {
+		try {
+			ResultSet rs = stmt.executeQuery( "SELECT * FROM CONVERSATION;");
+			while (rs.next() ) {
+				Uuid id = Uuid.fromString(rs.getString("id"));
+				String title = rs.getString("title");
+				Uuid owner = Uuid.fromString(rs.getString("owner"));
+				Time time = Time.fromMs(rs.getLong("creationtime"));
+
+				controller.newConversation(id, title, owner, time);
+			}
+			rs.close();
+			stmt.close();
+		}
+		catch ( Exception e ) {
+      		System.err.println( e.getClass().getName() + ": " + e.getMessage() );
+      		System.exit(0);
+   		}
+	}
+
+	public void createMessageTable() {
+		try {
+			String sql = "CREATE TABLE IF NOT EXISTS MESSAGE " + 
+					 "(ID             TEXT      NOT NULL," + 
+					 " AUTHOR         TEXT      NOT NULL," + 
+					 " CONVERSATION   TEXT      NOT NULL," + 
+					 " BODY           TEXT      NOT NULL," + 
+					 " CREATIONTIME   INTEGER      NOT NULL)";
+		stmt.executeUpdate(sql);
+		stmt.close();
+		}
+		catch (Exception e) {
+			System.err.println( e.getClass().getName() + ": " + e.getMessage() );
+			System.exit(0);
+		}
+	}
+
+	public void insertMessageTable(String id, String author, String convo, String body, Long time) {
+		try {
+			String sql = "INSERT INTO MESSAGE (ID, AUTHOR, CONVERSATION, BODY, CREATIONTIME)" + 

```

Comment:
This is an example where SQL Injection would be possible. 😮 
Using `PreparedStatements` is the solution here. If you'd like to learn more: https://docs.oracle.com/javase/tutorial/jdbc/basics/prepared.html